### PR TITLE
4 create new york transit data api extraction code

### PR DESCRIPTION
**Pull request by JPMartinezClaeys**
Sorry, this was a bit late but I did a last refactoring of the NYC code (I think it looks way better now).
The ingestion part has not been tested yet so that is pending but I wanted to do this PR so you can approve the code logic an ideally ingest this data tomorrow if everything is running ok.

I think the tests now will fail because it will try to import gdal stuff and it will raise an error when collecting data but here is a screenshot of the tests passing after I comment out every gdal-related item.
Screen Shot 2024-05-05 at 6 29 11 PM

### CODE REVIEW

File: backend/poetry.lock
Comment (meganhmoore) [inline backend/poetry.lock:1]: this whole folder should no longer exist

**Comment by meganhmoore** [comment]
> Sorry, this was a bit late but I did a last refactoring of the NYC code (I think it looks way better now). The ingestion part has not been tested yet so that is pending but I wanted to do this PR so you can approve the code logic an ideally ingest this data tomorrow if everything is running ok.
> 
> I think the tests now will fail because it will try to import gdal stuff and it will raise an error when collecting data but here is a screenshot of the tests passing after I comment out every gdal-related item.

I think this is due to the fact that we need to run anything that now hooks into django from the django test suite (https://github.com/uchicago-capp-30320/RouteRangers/blob/main/app/route_rangers_api/tests.py), so for now its ok if they are failing but will need to migrate tests there once ingestion is finalized

File: app/scripts/extract_nyc_data.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        "OBS_LEVEL": "bus_route",
+    },
+    "SUBWAY_RIDERSHIP": {
+        "URL": "https://data.ny.gov/resource/wujg-7c2s.json?",
+        "COLS_TO_KEEP": ["date", "station_complex_id", "ridership", "transfers"],
+        "GROUP_BY": "date, station_complex_id",
+        "OBS_LEVEL": "station_complex_id",
+    },
+}
+
```
Comment (meganhmoore) [inline app/scripts/extract_nyc_data.py:44]: do these come in as UTC and if so do you want to keep them that way since that is how the db is expecting?

**Review by meganhmoore — APPROVED** [review]
Once you remove the `backend/` folder all other comments can be addressed with the final ingestion to db so good with this for now.

**Review by jimenasalinas — APPROVED** [review]
Looks goof to merge! For future refactoring, we should try to separate ingestion from extraction until we figure out how to make our tests work (but not urgent)

**Review by meganhmoore — APPROVED** [review]
looks good w time zone issue created

**Pull request merged**